inpage/demo/App.tsx

```diff
@@ -15,6 +17,18 @@ const App = () => {
   useEffect(() => {
     // eslint-disable-next-line @typescript-eslint/no-floating-promises
     (async () => {
+      if (!address) {
+        const accounts = z.array(z.string()).parse(
+          await demo.ethereum.request({
+            method: 'eth_accounts',
```

Comment:
To clarify the difference between `eth_accounts` and `eth_requestAccounts`, `eth_accounts` only returns an array of **already connected** accounts, whereas `eth_requestAccounts` requests permission from the user to get all accounts (assuming the accounts not connected)?

Reply:
Pretty much. A couple of notes about `eth_requestAccounts` though:

- If any accounts are already connected, then `eth_requestAccounts` will just return those accounts without re-prompting (this might be just a convention, the requirement is just that `eth_requestAccounts` returns a non-empty array if it succeeds)
  - (In other words `eth_requestAccounts` only prompts when needed)
- `eth_requestAccounts` is only asking for access to accounts that the user is willing to grant, which is usually a single account, it's not asking for *all* accounts